resource.address/udp/src/test/java/org/kaazing/gateway/resource/address/udp/UdpResourceAddressFactorySpiPreferIPV4Test.java

```diff
@@ -63,4 +69,29 @@ public void startupWhenMatchingIPv6AddressesFound() throws Exception {
         factory.newResourceAddress("udp://[::1]:8000");
     }
 
+    @Test
+    public void shouldNotResolveLoopbackToIPv6IfProhibited() throws Exception {
+        // set the IPV4 flag to true
+        System.setProperty(JAVA_NET_PREFER_IPV4_STACK, "true");
```

Comment:
@jitsni @sbadugu Hello, Jitu, performing a debug when running the UdpResourceAddressFactorySpiPreferIPV4Test will reveal the following. Listing below methods:
- shouldNotResolveLoopbackToIPv6IfProhibited: in UdpResourceAddressFactorySpi, method newResourceAddresses0, line 171, 

>
boolean preferIPv4 = "true".equalsIgnoreCase(System.getProperty(JAVA_NET_PREFER_IPV4_STACK));

boolean preferIPv4 will be set to true since `System.getProperty(JAVA_NET_PREFER_IPV4_STACK)` will explicitly return the value set in the test set up section, i.e.:

>
System.setProperty(JAVA_NET_PREFER_IPV4_STACK, "true");

- shouldResolveLoopbackToIPv6IfNotProhibited: in UdpResourceAddressFactorySpi, method newResourceAddresses0, line 171, 

>
boolean preferIPv4 = "true".equalsIgnoreCase(System.getProperty(JAVA_NET_PREFER_IPV4_STACK));

boolean preferIPv4 will be set to false since `System.getProperty(JAVA_NET_PREFER_IPV4_STACK)` will explicitly return the value set in the test set up section, i.e.:

>
System.setProperty(JAVA_NET_PREFER_IPV4_STACK, "false");

When preferIPv4 is set to true the IPv6 filtering is performed at line 172, when it is set to false IPv6 addresses are also considered in the final result.